src/ResourceManager/Profile/Commands.Profile/Context/GetAzureRMContext.cs

```diff
@@ -26,6 +27,26 @@ namespace Microsoft.Azure.Commands.Profile
     [OutputType(typeof(PSAzureContext))]
     public class GetAzureRMContextCommand : AzureRMCmdlet
     {
+        /// <summary>
+        /// Gets the current default context.
+        /// </summary>
+        protected override AzureContext DefaultContext
+        {
+            get
+            {
+                if (DefaultProfile == null || DefaultProfile.Context == null)
+                {
+                    WriteError(new ErrorRecord(
+                        new PSInvalidOperationException("Run Login-AzureRmAccount to login."),
+                        string.Empty,
+                        ErrorCategory.AuthenticationError,
+                        null));
+                }
+
+                return DefaultProfile.Context;
```

Comment:
@shahabhijeet here is a comment Mark left on the original issue regarding this:

_In general, not having a context is a terminating error - meaning that any cmdlet pipeline should be stopped immediately regardless of error settings. In this particular case, since the work of the cmdlet is just getting the context, I agree that ErrorActionPreference should apply in this narrow case._